### ISSUE TITLE
Nx 18 Inferred Tasks support

### ISSUE THREAD

**Issue by Phillip9587** [Phillip9587/nx-stylelint#635]
Support Nx Project Crystal / [Inferred Tasks](https://nx.dev/concepts/inferred-tasks)

**Comment by Phillip9587** [comment]
**[v18.0.0-beta.3](https://github.com/Phillip9587/nx-stylelint/releases/tag/v18.0.0-beta.3) provides support for Nx 18 Inferred Tasks**

Please test it out and provide feedback. 

To add the plugin add the following to your `nx.json`:

```json
{
    "plugins": [
        {
            "plugin": "nx-stylelint/plugin",
            "options": {
                "targetName": "stylelint",
                "extensions": ["css"]
            }
        }
    ]
}
```
targetDefaults can be configured e.g. `--allow-empty-input`: 

```json
"targetDefaults": {
    "stylelint": {
        "options": {
            "args": ["--allow-empty-input"]
        }
    }
}
```

**Comment by Jesse1989pp** [comment]
@Phillip9587 Amazing, it seems to infer the tasks properly for me. Only issue i am encountering at the moment is extending my config from a generic `node_module`. I use a base config from my company which includes `stylelint-config-recommended-scss`. I am not able to reach this now, which triggers an error. 

When i try and reach it relative, a read error is triggered via `cosmiconfig`. However not sure if either of these errors are `cosmiconfig` related yet.

**Comment by mateusgpereira** [comment]
I add the plugin to the nx.json file:
```json
{
   "plugin": "nx-stylelint/plugin",
    "options": {
        "targetName": "stylelint",
        "extensions": ["scss"]
    }
 }
```

I'm also using this on targetDefaults:
```json
"targetDefaults": {
  "stylelint": {
    "inputs": [
      "default",
      "{workspaceRoot}/.stylelintrc(.(json|yml|yaml|js))?"
    ],
    "cache": true,
    "options": {
      "args": [
        "--allow-empty-input"
      ]
    }
  }
}
```

With this setup I can see the task being inferred in nx console with this description:  
`Created by nx/core/target-defaults from nx.json`  
but only for the libraries that have their own `.stylelintrc` file.  
It would be nice to have this task inferred independently of the lib having or not its own configuration file and using the rules defined at the workspaceRoot level.

Great job with this plugin guys ^^

**Comment by Phillip9587** [comment]
[nx-stylelint v18](https://github.com/Phillip9587/nx-stylelint/releases/tag/v18.0.0) includes the experimental plugin. The generators do not support adding the plugin yet but it is planned once Nx exports the needed helpers from `@nx/devkit`. Please test it out and provide feedback.
